app/components/signature-pill.js

```diff
@@ -11,84 +11,49 @@ export default class SignaturePillComponent extends Component {
-      return "error";
-    } else if (this.signingHubUrl) {
-      return "link"
-    } else {
-      return "ongoing"
-    }
-  }
+    return await (await signSubcase.signFlow).status;
+  });
 
-  get title() {
-    if (this.isCancelled) {
-      return this.intl.t('cancelled');
-    } else if (this.isRefused) {
-      return this.intl.t('refused');
-    } else if (this.isSigned) {
-      return this.intl.t('signed');
-    } else if (this.hasToBeSigned) {
-      return this.intl.t('to-sign');
-    } else if (this.hasToBeApproved) {
-      return this.intl.t('to-approve');
-    } else if (this.isPrepared) {
-      return this.intl.t('sent');
-    } else if (this.isMarked) {
-      return this.intl.t('to-sign');
+  signingHubUrl = trackedFunction(this, async () => {
+    const { SIGNED, REFUSED } = constants.SIGNFLOW_STATUSES;
+    if (!this.status.value || this.status.value.uri === REFUSED) {
+      return null;
     }
-    return "";
-  }
 
-  loadData = task(async () => {
+    const currentUser = this.currentSession.user;
+    const status = this.status.value;
+    const pieceArg = this.args.piece;
     const signMarkingActivity = await this.args.signMarkingActivity;
-    if (signMarkingActivity) {
-      const signSubcase = await signMarkingActivity.signSubcase;
-      const signPreparationActivity = await signSubcase.signPreparationActivity;
-      const signApprovalActivities = await signSubcase.signApprovalActivities;
-      const signSigningActivities = await signSubcase.signSigningActivities;
-      const signCompletionActivity = await signSubcase.signCompletionActivity;
-      const signRefusalActivities = await signSubcase.signRefusalActivities;
-      const signCancellationActivity = await signSubcase.signCancellationActivity;
-
-      this.isMarked = !!signMarkingActivity;
-      this.isPrepared = !!signPreparationActivity;
-      this.hasToBeApproved = signApprovalActivities?.some((activity) => activity.startDate && !activity.endDate);
-      this.hasToBeSigned = signSigningActivities?.some((activity) => activity.startDate && !activity.endDate);
-      this.isSigned = !!signCompletionActivity;
-      this.isRefused = signRefusalActivities?.length;
-      this.isCancelled = !!signCancellationActivity;
-
-      if (!this.isRefused) {
-        const piece = await this.args.piece;
-        const signFlow = await signSubcase.signFlow;
-        const signFlowCreator = await signFlow.creator;
-        const currentUser = this.currentSession.user;
-        if (piece && (signFlowCreator.id === currentUser.id) && !this.isSigned) {
-          const response = await fetch(
-            `/signing-flows/${signFlow.id}/pieces/${piece.id}/signinghub-url?collapse_panels=false`
-          );
-          if (response.ok) {
-            const result = await response.json();
-            this.signingHubUrl = result.url;
-          } 
-        }
+    const signSubcase = await signMarkingActivity.signSubcase;
+    const piece = await pieceArg;
```

Comment:
Why not `await this.args.piece` directly?

Reply:
This was indeed not necessary, was a leftover of some experimentation with `trackedFunction`